dianna/utils/maskers.py

```diff
@@ -14,9 +14,39 @@ def generate_masks(input_data: np.array, number_of_masks: int, p_keep: float = 0
     Returns:
     Single array containing all masks where the first dimension represents the batch.
     """
-    series_length = input_data.shape[0]
-    number_of_steps_masked = _determine_number_of_steps_masked(p_keep, series_length)
+    if input_data.shape[-1] == 1:  # univariate data
+        return generate_time_step_masks(input_data, number_of_masks, p_keep)
+
+    number_of_channel_masks = number_of_masks // 3
+    number_of_time_step_masks = number_of_channel_masks
+    number_of_combined_masks = number_of_masks - number_of_time_step_masks - number_of_channel_masks
```

Comment:
This is quite an interesting way to implement the masking for multi-channels. I can understand that by doing this we can have a very balanced masking array, which contains masking for entire channels, masking for certain time steps across channels and a mixture of them. But I have a feeling that this makes it a bit too complex.

What I have in mind are two simple ways:
1. Simple flatten the input data and mask them brute-forcely
2. Loop through all channels and treat them individually (mask each channel separately and concatenate them)

I can imagine that the current implementation makes the segmentation very tricky to code. Let's have a chat about it. It could be possible that I misunderstand something.

But thanks a lot for the effort! This also provides more insight about what we want.